utils/utils.go

```diff
@@ -0,0 +1,478 @@
+	ProcsRegexp = `(%s-v\d+-[\w-]+) up \(v\d+\)`
+
+	errMissingRouterHostEnvVar = fmt.Errorf("missing %s", deisRouterServiceHost)
+	DefaultMaxTimeout          = 5 * time.Minute // gomega's default is 2 minutes
+	Url                        = getController()
+	Debug                      = os.Getenv("DEBUG") != ""
+	HomeHome                   = os.Getenv("HOME")
+)
+
+func Cancel(url, username, password string) {
+	// log in to the account
+	Login(url, username, password)
+
+	// remove any existing test-* apps
+	sess, err := Run("deis apps")
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	re := regexp.MustCompile("test-.*")
+	for _, app := range re.FindAll(sess.Out.Contents(), -1) {
+		sess, err = Run("deis destroy --app=%s --confirm=%s", app, app)
+		Expect(err).To(BeNil())
+		Eventually(sess).Should(Say("Destroying %s...", app))
+		Eventually(sess).Should(Exit(0))
+	}
+
+	// cancel the account
+	sess, err = Run("deis auth:cancel --username=%s --password=%s --yes", username, password)
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Account cancelled"))
+}
+
+func Login(url, user, password string) {
+	sess, err := Run("deis login %s --username=%s --password=%s", url, user, password)
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Logged in as %s", user))
+}
+
+func Logout() {
+	sess, err := Run("deis auth:logout")
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Exit(0))
+	Eventually(sess).Should(Say("Logged out\n"))
+}
+
+func Run(cmdLine string, args ...interface{}) (*Session, error) {
+	ourCommand := Cmd{Env: os.Environ(), CommandLineString: fmt.Sprintf(cmdLine, args...)}
+	return startCmd(ourCommand)
+}
+
+func startCmd(command Cmd) (*Session, error) {
+	cmd := exec.Command("/bin/sh", "-c", command.CommandLineString)
+	cmd.Env = command.Env
+	io.WriteString(GinkgoWriter, fmt.Sprintf("$ %s\n", command.CommandLineString))
+	return Start(cmd, GinkgoWriter, GinkgoWriter)
+}
+
+func register(url, username, password, email string) {
+	sess, err := Run("deis register %s --username=%s --password=%s --email=%s", url, username, password, email)
+	Expect(err).To(BeNil())
+	Eventually(sess).Should(Say("Registered %s", username))
+	Eventually(sess).Should(Say("Logged in as %s", username))
+}
+
+func RegisterOrLogin(url, username, password, email string) {
+	sess, err := Run("deis register %s --username=%s --password=%s --email=%s", url, username, password, email)
+
+	Expect(err).To(BeNil())
+
+	sess.Wait()
+
+	if strings.Contains(string(sess.Err.Contents()), "must be unique") {
+		// Already registered
+		Login(url, username, password)
+	} else {
+		Eventually(sess).Should(Exit(0))
+		Eventually(sess).Should(SatisfyAll(
+			Say("Registered %s", username),
+			Say("Logged in as %s", username)))
+	}
+}
+
+// execute executes the command generated by fmt.Sprintf(cmdLine, args...) and returns its output as a cmdOut structure.
```

Comment:
nit: now that some of these methods have now been capitalized (officially public), the godoc description (comments) should be updated as well.